File: 4-陈浩浩-深圳/第四周/canny.cpp
#include<iostream>
#include<opencv2/opencv.hpp>
#include<string>

using namespace cv;
using namespace std;

int main(int argc, char* argv[]){
    string imgPath = argv[1];
    Mat img = imread(imgPath);
    if(!img.data){
        cout << "loading image failed: " << imgPath << endl;
        return -1;
    }
    Mat edges;
    Canny(img, edges, 30, 200);

    imshow("img", img);
    imshow("edges", edges);
    waitKey(0);
    destroyAllWindows();

    return 0;
}

//void Canny(cv::InputArray image, cv::OutputArray edges, double threshold1, double threshold2, int apertureSize = 3, bool L2gradient = false)
// threshold1: 低阈值
// threshold2: 高阈值
// apertureSize：sobel算子大小，默认采用3*3
// L2gradient：true是采用L2方式计算梯度(更精确), false时采用L1

File: 61-桂志鹏-北京/week4/canny.py
# canny实现
import cv2

img = cv2.imread('img/lenna.png', 1)
cv2.imshow('img', img)
gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
can = cv2.Canny(gray, 200, 300)
cv2.imshow('canny', can)

cv2.waitKey(0)
cv2.destroyAllWindows()
